### PR TITLE
Bug 1845255 - Add loading UI for review quality check

### PR DESCRIPTION
**Pull request by rahulsainani**
– Figma and animation css in BZ. 

### Screen Recording
https://github.com/mozilla-mobile/firefox-android/assets/38040960/5aadcd0d-6b80-4bb0-9f7b-382a622eac05

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1845255

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductReviewLoading.kt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.theme.FirefoxTheme
+
+private const val ANIMATION_DURATION_MS = 1000
+private const val NUMBER_OF_BOXES = 5
+private const val INITIAL_ALPHA = 0.25f
+private const val TARGET_ALPHA = 1f
+private val boxHeights = listOf(80.dp, 80.dp, 192.dp, 40.dp, 192.dp)
```
Comment (t-p-white) [inline fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductReviewLoading.kt:35]: What are your thoughts on having these values as named consts aswell? E.g 
```
RELIABLE_BOX_HEIGHT = 80.dp, 
ADJUSTED_RATINGS_BOX_HEIGHT = 80.dp 
```
etc
Same comment for the animation list below

Comment (t-p-white) [reply fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductReviewLoading.kt:35]: Or having a data class such as BoxDesc/Info and having 1 list and a forEach 🙂

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductReviewLoading.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+/**
+ * Loading UI for review quality check content.
+ */
+@Composable
+fun ProductReviewLoading() {
+    val infiniteTransition = rememberInfiniteTransition("ProductReviewLoading")
+
+    Column(
+        modifier = Modifier.padding(0.dp),
```
Comment (t-p-white) [inline fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductReviewLoading.kt:46]: nit: think we can remove setting the padding

**Review by t-p-white — COMMENTED** [review]
This looks great @rahulsainani 🙌

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductReviewLoading.kt (hunk 1 above)
Comment (rahulsainani) [reply fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductReviewLoading.kt]: Yeah, good point! I agree about the `BoxInfo` and having one list.

I think extracting/naming each height/offset is not necessary now with the `BoxInfo`, it's more clear what they mean. wdyt?

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductReviewLoading.kt (hunk 2 above)
Comment (rahulsainani) [reply fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductReviewLoading.kt]: good catch, updated and added modifier default param for `ProductReviewLoading`.

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductReviewLoading.kt (hunk 1 above)
Comment (t-p-white) [reply fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductReviewLoading.kt]: Agreed, no need to extract the height/offsets 🙂

**Review by t-p-white — APPROVED** [review]
lgtm :shipit:

**Pull request merged**